packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -1578,9 +1590,15 @@ function pushScript(
   props: Object,
   responseState: ResponseState,
   textEmbedded: boolean,
+  insertionMode: InsertionMode,
   noscriptTagInScope: boolean,
 ): null {
-  if (enableFloat && !noscriptTagInScope && resourcesFromScript(props)) {
+  if (
+    enableFloat &&
+    insertionMode !== SVG_MODE &&
```

Comment:
I considered also checking for MathML mode but I don't think there are any tag collisions in that namespace and it would require an additional branch or tracking an aggregate namepsace. We may need that someday but I think this suffices for now